### PR TITLE
Ss/refine dex

### PR DESCRIPTION
**Pull request by schaeferCode**
## Ticket
* none

## Changes
* Hooked up live data. Not everything is done, though. 
* Enabled list filter. 
* Enabled pair switching.

## Screenshots
screen shot 2018-09-24 at 6 03 53 pm

screen shot 2018-09-24 at 6 04 16 pm

screen shot 2018-09-24 at 6 04 37 pm

### CODE REVIEW

File: src/components/SimpleTable.vue [hunk 1]
```diff
@@ -12,15 +12,18 @@
           </th>
         </tr>
       </thead>
-      <tbody class="table-body-wrapper">
-        <tr class="row" v-for="entry in filteredData">
+      <tbody class="table-body-wrapper" v-if="data.length > 0">
+        <tr class="row" v-for="entry in filteredData" @click="handleRowClick ? handleRowClick(entry) : () => {}">
```
Comment (crishellco) [inline src/components/SimpleTable.vue]: Would this maybe be cleaner if you abstracted it into a method?

File: src/components/dex/Pair.vue [hunk 2]
```diff
@@ -20,50 +20,104 @@
...
+  mounted() {
+    this.loadTrades();
+    this.baseCurrency = _.first(this.baseCurrencies);
+
+    storeUnwatch = this.$store.watch(
+      () => {
+        return this.$store.state.currentMarket;
+      }, () => {
+        this.loadTrades();
+      });
```
Comment (schaeferCode) [inline src/components/dex/Pair.vue:120]: @crishellco, This is what I was asking about this last weekend. I'm not sure what this is doing. It seems like it's setting up a session of 'watching' a store value. Why can't this be done in the watch methods?

**Review by crishellco — APPROVED** [review]

File: src/services/dex.js [hunk 3]
```diff
@@ -195,14 +117,35 @@ export default {
             resolve(history);
           })
           .catch((e) => {
-            alerts.exception(`APH API Error: ${e}`);
+            alerts.exception(`APH API Error: ${e.message}`);
           });
       } catch (e) {
-        reject(`Failed to fetch trade history. ${e.message}`);
+        reject(e);
```
Comment (schaeferCode) [inline src/services/dex.js:123]: Also code I didn't touch.

File: src/components/dex/Pair.vue (hunk 2 above)
Comment (crishellco) [reply src/components/dex/Pair.vue:120]: Yeah, so this _should_ work with a normal `watch`

```
computed: {
  ...mapGetters(['currentMarket']),
},
```

File: src/components/SimpleTable.vue (hunk 1 above)
Comment (schaeferCode) [reply src/components/SimpleTable.vue]: Yeah for sure.

File: src/components/dex/Pair.vue (hunk 2 above)
Comment (schaeferCode) [reply src/components/dex/Pair.vue:120]: I don't understand what is going on here. I'm not sure these equal each other. I'm leaving it as is, but feel free to mess with it in another PR.

**Pull request merged**